packages/fxa-payments-server/src/components/PaymentForm/index.tsx

```diff
@@ -362,6 +379,22 @@ export const PaymentForm = ({
     </SubmitButton>
   );
 
+  if (logGAEvent === false) {
+    if (
+      validator.state.fields.name?.value &&
+      validator.state.fields.creditCard?.value?.complete &&
+      validator.allValid()
+    ) {
```

Comment:
Since these are nested `if`s with no `else`, does it make sense to just combine them and remove a level of complexity?
```js
    if (
      logGAEvent === false &&
      validator.state.fields.name?.value &&
      validator.state.fields.creditCard?.value?.complete &&
      validator.allValid()
    ) {
```

But now that I type that silliness, what is this doing? L382 says "if logGAEvent is false" (plus some nested conditionals) then we call `ReactGALog.logEvent(...)`?
I'm guessing `logGAEvent` here isn't referring to whether we SHOULD log GA events, and more whether an event has been logged yet?

Reply:
I was looking to prevent the same event being logged multiple times, as it would if any changes were made after all fields were either filled in/completed/validated.